### DISCUSSION TITLE
Unable set default project in vscode

### DISCUSSION THREAD

**Discussion by sarosigy** [unoplatform/uno#15475] [Q&A] [answered]
### Current behavior

Freshly created uno project:
dotnet new unoapp -sw:false -mobile:false -slf:false -vs -s -wpm -ho -f:net7.0 
When try to set the default project to Skia.GTK or WASM, the following error message appears:

Project.Skia.Gtk can only be built using .NET 7 when using Uno.WinUI 5.0.116 or later. Update your version of Uno or update to .NET 8.

It is unclear to me what the error message wants to say. My concerns are:
1. 'can only be built using .NET 7': I have only .NET 7
2. 'using Uno.WinUI 5.0.116 or later': the referenced assemblies are 4.7.56, not even near to 5.0.116 (default from solution templates)
3. 'Update your version of Uno': what to update? The solution templates are the newest, the vscode extension is also the newest. Is it refers to the assemblies?
4. 'update to .NET 8': If 'can only be built using .NET 7' why should I update to .NET 8?

### Expected behavior

1. Please, Clarify the error message (exactly what to update and to which version)
2. If it is possible, make a chart with the working set of  .net / Uno assemblies / extension / solution templates

### How to reproduce it (as minimally and precisely as possible)

0. .NET 7 installed on debian linux
1. dotnet new unoapp -sw:false -mobile:false -slf:false -vs -s -wpm -ho -f:net7.0 
2. Try to set the default project to Skia.GTK or WASM

### Works on UWP/WinUI

None

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)
```
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageReference Include="Uno.WinUI" Version="4.7.56" />
  </ItemGroup>
  <ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.18362'">
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.3" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.22000.24" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.22000.24" />
  </ItemGroup>
```
And
```
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.WinUI.Skia.Gtk" Version="4.7.56" />
    <PackageReference Include="Uno.WinUI.RemoteControl" Version="4.7.56" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.7.56" />
    <PackageReference Include="Uno.WinUI.Lottie" Version="4.7.56" />
    <PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.3" />
    <PackageReference Include="SkiaSharp.Skottie" Version="2.88.3" />
  </ItemGroup>
```

### Affected platforms

Build tasks

### IDE

Visual Studio Code

### IDE version

1.86.1

### Relevant plugins

C# v2.18.15 (pre-release)
C# Dev Kit v1.3.10
Uno Platform v0.12.0

### Anything else we need to know?

Debian linux bookworm
.Net 7 installed

**Comment by jeromelaban** [comment, marked as answer]
Thanks for your question. 

This message means that you either need to update to .NET 8 if you want to keep using your current version of Uno, or you need to update your version of Uno to 5.0.116 to keep using .NET 7.

**Reply by jeromelaban** [reply]
> So, If I'm correct then the 4.7.56 requires .NET 8 to use the vscode extension (for example XAML intellisense), but 5.0.116 and later would allow .NET 7 also?

That is correct.

**Comment by sarosigy** [comment]
So, If I'm correct then the 4.7.56 requires .NET 8 to use the vscode extension (for example XAML intellisense), but 5.0.116 and later would allow .NET 7 also?

**Comment by sarosigy** [comment]
OK, understand now, and using the newest of everything.
